src/unfolding_tests/getBestTau.py

```diff
@@ -0,0 +1,451 @@
+        channel = regularisation_settings.channel
+        com = regularisation_settings.centre_of_mass_energy
+        print 'Variable = {0}, channel = {1}, sqrt(s) = {2}'.format(variable, channel, com)
+
+        # Find the corresponding Chi2
+        pd_chi2, pd_taus = get_chi2s_of_tau_range(regularisation_settings, args)
+
+        # Write to file
+        df_chi2 = chi2_to_df(pd_chi2, pd_taus, regularisation_settings) 
+
+    # Have the dataframes now - albeit read to a file
+    # Read in each one corresponding to their channel
+    # Find the best tau and pront to screen
+    # Maybe find way to include in previous loop (would requires three dataframes instantiations)
+    for channel in ['electron', 'muon', 'combined']:
+        path = regularisation_settings.outpath+'tbl_'+channel+'_tauscan.txt'
+        df_chi2 = get_df_from_file(path)
+        if df_chi2 is None: continue
+        print '\n', "1 - P(Chi2|NDF)", '\n', df_chi2, '\n'
+
+        # cutoff to be changed to 0.001 when able to
+        best_taus = interpolate_tau(0.3, df_chi2)
+        chi2_to_plots(df_chi2, regularisation_settings, channel)
+        print_results_to_screen(best_taus, channel)
+
+
+def parse_options():
+    '''
+    parse the config jsons from command line and read the contents of the json files
+    '''
+    parser = ArgumentParser( __doc__ )
+    parser.add_argument("in_files",
+        nargs='*',
+        help="List of the input files")
+    parser.add_argument( "-t", "--test", 
+        dest = "run_measured_as_data",
+        action = "store_true", 
+        help = "For debugging - run the measured distribution as data." ) 
+    parser.add_argument( "-p", "--refold_plots", 
+        dest = "run_refold_plots",
+        action = "store_true", 
+        help = "For debugging - output unfolded vs refolded for each tau" ) 
+    parser.add_argument( "-n", "--n_taus", 
+        dest = "n_tau_scan_points",
+        default = 25,
+        type = int,
+        help = "How many taus in the range do you want" ) 
+
+    args = parser.parse_args()
+
+    input_values_sets = []
+    json_input_files = []
+    add_set = input_values_sets.append
+    add_json_file = json_input_files.append
+    for arg in args.in_files:
+        input_values = read_data_from_JSON( arg )
+        add_set( input_values )
+        add_json_file( arg )
+
+    return args, input_values_sets, json_input_files
+
+def clear_old_df(path):
+    '''
+    Delete any previous dataframe. (Code would append a new dataframe to file instead of replace)
+    '''
+
+    for root, dirs, files in os.walk(path, topdown=False):
+        for name in files:
+            os.remove(os.path.join(root, name))
+    return
+
+def get_tau_list(n_tau):
+    '''
+    Large scanning range from unity to 10^-8. Split into equal points based on log system 
+    given the number of tau points to scan over.
+    '''
+    tau_test_range = [pow(10, -8*i/n_tau) for i in range(0, n_tau)]
```

Comment:
Odd - it was definitely working fine for me. Either way I've now implemented this tau choice function.  Presently it runs run max tau to min tau as that was the way it was set up before. I will go back at somepoint and fix things such that it goes more intuitively from min tau to max tau